### ISSUE TITLE
import-name rule renaming map is not working

### ISSUE THREAD

**Issue by alexeychikk** [microsoft/tslint-microsoft-contrib#881]
### Bug Report

-   **`tslint-microsoft-contrib` version**: 6.2.0
-   **TSLint version**: 5.18.0
-   **TypeScript version**: 3.5.2
-   **Running TSLint via**: VSCode

#### TypeScript code being linted

```tsx
import React from 'react';
```

with `tslint.json` configuration:

```json
{
  "extends": "tslint-microsoft-contrib",
  "rules": {
    "import-name": [
      true,
      { "react": "React" }
    ]
  }
}

```

#### Actual behavior
`import React` is being replaced with `import react` on save.
 
#### Expected behavior
Nothing should happen because I defined `{ "react": "React" }` in my tslint.json.

**Comment by dosentmatter** [comment]
Is your config extending anything else? I had the same issue but I found that the tslint-config-airbnb@5.11.1 I was using depended on tslint-microsoft-contrib@5.2.1.

The docs for `import-name` say

>Since version 2.0.9 it is possible to configure this rule with a list of exceptions.

But perhaps they had a bug in version 5.2.1.

Anyway, to test, I installed version tslint-microsoft-contrib@6.2.0 and did `rm -rf node_modules/tslint-config-airbnb/node_modules/tslint-microsoft-contrib`, to force it to resolve to version 6.2.0.

I got some deprecated rules as expected from jumping up a major version, but the `import-name` error went away. So, I think it's an issue with some other config you are extending and maybe not tslint-microsoft-contrib. I will create an issue on tslint-config-airbnb.

---

If you have the above issue, and if you want to hack it (in a reproducible way) to force it to resolve to 6.2.0, you can do the following:
```sh
npm install tslint-microsoft-contrib # get 6.2.0
```

Now you can add `tslint-microsoft-contrib` to `rulesDirectory`, but from what I see, the way tslint works is that it `requires` sub-dependencies before direct dependencies.
In other words, `require('tslint-config-airbnb/node_modules/tslint-microsoft-contrib')` will happen before `require('tslint-microsoft-contrib')`, because `tslint-contrib-airbnb` is inside `extends`, so airbnb's tslint-microsoft-contrib is required first.

You can see it in the output of `tslint --print-config`:
```json
{
  "rulesDirectory": ["tslint-microsoft-contrib"],
  "extends": ["tslint-config-airbnb"]
}
```

becomes

```json
  "rulesDirectory": [
    ".../node_modules/tslint-consistent-codestyle/rules",
    ".../node_modules/tslint-eslint-rules/dist/rules",
    ".../node_modules/tslint-config-airbnb/node_modules/tslint-microsoft-contrib",
    ".../node_modules/tslint-microsoft-contrib"
  ]
```

That doesn't work since `tslint-config-airbnb/node_modules/tslint-microsoft-contrib` comes before `tslint-microsoft-contrib`.

---

If you did:
```json
{
  "extends": ["tslint-microsoft-contrib", "tslint-config-airbnb"]
}
```
Note that the opposite order, `"extends": ["tslint-config-airbnb", "tslint-microsoft-contrib"]`, wouldn't work because `"tslint-microsoft-contrib"` would get required second.

you get the right thing

```json
  "rulesDirectory": [
    ".../node_modules/tslint-microsoft-contrib",
    ".../node_modules/tslint-consistent-codestyle/rules",
    ".../node_modules/tslint-eslint-rules/dist/rules",
    ".../node_modules/tslint-config-airbnb/node_modules/tslint-microsoft-contrib"
  ]
```

The problem with this method is that it extends all of the [base rules](https://npm.runkit.com/tslint-microsoft-contrib/tslint.json?t=1563484429418). I don't want any `rules`, I just want the `rulesDirectory`.

---

So the final workaround is this:
```json
{
  "extends": ["./tslint-microsoft-contrib.json", "tslint-config-airbnb"]
}
```

In `./tslint-microsoft-contrib.json`, just have the `rulesDirectory`:
```
{
  "rulesDirectory": ["tslint-microsoft-contrib"]
}
```

Now you will get the correct `require` order to override and you won't extend any `rules`.

**Comment by alexeychikk** [comment]
@dosentmatter Well, thanks for your feedback. But the config I specified in the bug report is the actual config that makes this bug reproducible. I don't extend in my `tslint.json` any other configs, just `tslint-microsoft-contrib`. And taking into account referenced issues seems like the bug is there.

**Comment by dosentmatter** [comment]
@alexeychikk, hmm, so the PR #882 that references this issue was created by me. I used the latest version 6.2.0, as you did.

I fixed a [few bugs and issues](https://github.com/microsoft/tslint-microsoft-contrib/pull/882#issue-299189760). Here are the ones relevant to you:

1. `import-name` has an off by one issue. The workaround right now is to pass in `null` as the first option:
```
  "rules": {
    "import-name": [
      true,
      null,
      { "react": "React" }
    ]
  }
```
2. There are actually a few more options that aren't explained in the README. I documented them in my [fork](https://github.com/dosentmatter/tslint-microsoft-contrib).

**Comment by alexeychikk** [comment]
@dosentmatter Thanks. After experimenting for a while I figured it out. This works for me well:
```json
 "rules": {
    "import-name": [
      true,
      null,
      {},
      null,
      { "ignoreExternalModule": false, "case": "any-case" }
    ],
  }
```

**Comment by dosentmatter** [comment]
@alexeychikk, no problem. Glad I could help. Remember to update it when the bug gets fixed!

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #876. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!
